### PR TITLE
Update generator to reflect GAX changes

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: Google.Api.Generator.Tests/ProtoTests/GrpcServiceConfig/Testing.GrpcServiceConfig/GrpcServiceConfigClient.g.cs [hunk 1]
```diff
@@ -34,13 +34,11 @@ public sealed partial class GrpcServiceConfigSettings : gaxgrpc::ServiceSettings
         /// <item><description>Initial retry delay: 500 milliseconds.</description></item>
         /// <item><description>Retry delay multiplier: 1.3</description></item>
         /// <item><description>Retry maximum delay: 5000 milliseconds.</description></item>
-        /// <item><description>Initial timeout: 20000 milliseconds.</description></item>
-        /// <item><description>Timeout multiplier: 1</description></item>
-        /// <item><description>Timeout maximum delay: 20000 milliseconds.</description></item>
-        /// <item><description>Total timeout: 20 seconds.</description></item>
+        /// <item><description>Maximum attempts: 2147483647</description></item>
```
Comment (chrisdunelm) [inline Google.Api.Generator.Tests/ProtoTests/GrpcServiceConfig/Testing.GrpcServiceConfig/GrpcServiceConfigClient.g.cs]: This should probably say "Maximum attempts: unlimited" or similar.

File: Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs [hunk 2]
```diff
@@ -108,42 +108,46 @@ IEnumerable<PropertyDeclarationSyntax> PerMethod(MethodDetails method)
                 var cSync = XmlDoc.C($"{_svc.ClientAbstractTyp.Name}.{method.SyncMethodName}");
                 var cAsync = XmlDoc.C($"{_svc.ClientAbstractTyp.Name}.{method.AsyncMethodName}");
                 // Add the general per-method settings property.
-                var property = AutoProperty(Public, _ctx.Type<CallSettings>(), method.SettingsName, hasSetter: true);
+                // We always specify an expiration, and the retry is optional.
+                var expiration = method.Expiration ?? Expiration.None;
+                var property = AutoProperty(Public, _ctx.Type<CallSettings>(), method.SettingsName, hasSetter: true)
+                    .WithInitializer(_ctx.Type<CallSettings>().Call(nameof(CallSettings.FromExpiration))(
+                        expiration.Type == ExpirationType.None
+                        ? _ctx.Type(typeof(Expiration)).Access(nameof(Expiration.None))
```
Comment (chrisdunelm) [inline Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs]: nit: Indent this and the next line?

File: Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs [hunk 3]
```diff
@@ -108,42 +108,46 @@ IEnumerable<PropertyDeclarationSyntax> PerMethod(MethodDetails method)
                 var cSync = XmlDoc.C($"{_svc.ClientAbstractTyp.Name}.{method.SyncMethodName}");
                 var cAsync = XmlDoc.C($"{_svc.ClientAbstractTyp.Name}.{method.AsyncMethodName}");
                 // Add the general per-method settings property.
-                var property = AutoProperty(Public, _ctx.Type<CallSettings>(), method.SettingsName, hasSetter: true);
+                // We always specify an expiration, and the retry is optional.
+                var expiration = method.Expiration ?? Expiration.None;
+                var property = AutoProperty(Public, _ctx.Type<CallSettings>(), method.SettingsName, hasSetter: true)
+                    .WithInitializer(_ctx.Type<CallSettings>().Call(nameof(CallSettings.FromExpiration))(
```
Comment (chrisdunelm) [inline Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs]: This initializer looks like it's overridden if the method has retry.
Would it be better to only apply this initializer in the `else` clause a bit further down?

File: Google.Api.Generator.Tests/ProtoTests/GrpcServiceConfig/Testing.GrpcServiceConfig/GrpcServiceConfigClient.g.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Generator.Tests/ProtoTests/GrpcServiceConfig/Testing.GrpcServiceConfig/GrpcServiceConfigClient.g.cs]: Done.

File: Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs (hunk 3 above)
Comment (jskeet) [reply Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs]: It's not so much overridden as built on. But I can probably do something cleaner.

File: Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs (hunk 2 above)
Comment (jskeet) [reply Google.Api.Generator/Generation/ServiceSettingsCodeGenerator.cs]: Done.

**Comment by chrisdunelm** [comment]
There's a couple of tests failing, but LGTM once those are sorted out.

**Comment by jskeet** [comment]
I'm very confused as to why the tests are failing. It looks like it's trying to build against .NET Standard 1.5, but I didn't think there was anything left targeting that. Hmm.

File: Google.Api.Generator.Tests/ProtoTests/Lro/Testing.Lro/Testing.Lro.csproj [hunk 4]
```diff
@@ -30,7 +30,7 @@
     -->
 
     <!-- These items should not require editing -->
-    <TargetFrameworks>netstandard1.5;net45</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
```
Comment (jskeet) [inline Google.Api.Generator.Tests/ProtoTests/Lro/Testing.Lro/Testing.Lro.csproj]: This is the line that changes the targets. So why was it running against the old ones?

**Pull request merged**